### DISCUSSION TITLE
Support for case-insensitive value checks

### DISCUSSION THREAD

**Discussion by andyward** [buildingSMART/IDS#346] [Ideas]
This has been discussed [before](https://github.com/buildingSMART/IDS/issues/30), but it's not clear to me why we have determined that **all** value comparisons must be case sensitive in IDS. E.g. test cases such as [fail-attributes_should_check_strings_case_sensitively_2_2.ids
](https://github.com/buildingSMART/IDS/blob/development/Documentation/ImplementersDocumentation/TestCases/attribute/fail-attributes_should_check_strings_case_sensitively_2_2.ids) and [fail-specifying_a_value_performs_a_case_sensitive_match_2_2.ids](https://github.com/buildingSMART/IDS/blob/development/Documentation/ImplementersDocumentation/TestCases/property/fail-specifying_a_value_performs_a_case_sensitive_match_2_2.ids)

While I can see there may be some scenarios where it makes sense when dealing with fixed schema values (e.g. #49) it feels like we're making life harder for both IDS and Model authors by not allowing some allowance for case differences - particularly on things like Attribute Values, PropertyValues, Classification values etc. where these are typically free text fields.

Take this simple example from a client's EIR/BEP, where they want to specify floors should have a consistent description:

``` xml
<ids:specification ifcVersion="IFC2X3" name="04.05 : Building Storeys should have standard Description">
		<ids:applicability minOccurs="1" maxOccurs="unbounded">
			<ids:entity>
				<ids:name>
					<xs:restriction base="xs:string">
						<xs:enumeration value="IFCBUILDINGSTOREY" />
					</xs:restriction>
				</ids:name>
			</ids:entity>
		</ids:applicability>
		<ids:requirements>
			<ids:attribute cardinality="required">
				<ids:name>
					<ids:simpleValue>Description</ids:simpleValue>
				</ids:name>
				<ids:value>
					<xs:restriction base="xs:string">
						<xs:enumeration value="Basement Level 2" />
						<xs:enumeration value="Basement Level 1" />
						<xs:enumeration value="Base level of building (Ground Floor)" />
						<xs:enumeration value="First Floor" />
						<xs:enumeration value="Second Floor" />
						<xs:enumeration value="Third Floor" />
					</xs:restriction>
				</ids:value>
			</ids:attribute>
		</ids:requirements>
</ids:specification>
```
I can see model authors questioning the value of being asked to change 'Basement level 1' to 'Basement Level 1' when there may be many other high-value issues in the information.  At the moment as an IDS author I'd have to express this as a Regex to allow some degree of tolerance.

Should we consider a case sensitivity option on values? Perhaps something like:

```xml
<ids:value stringComparison="ignoreCase">
  <xs:restriction base="xs:string">
    <xs:enumeration value="Basement Level 2" />
... etc
```
Obviously this would only take affect on string values, and I get there may be internationalisation concerns for Unicode etc (but that's a solved problem?)

**Comment by atomczak** [comment]
We are after consistent and qualitative data and 'Basement level 1' is not equal 'Basement Level 1'. I'm personally against case insensitivity.

**Reply by aothms** [reply]
Tend to agree, you can use a regex and be really explicit about it if needed.
